src/pytesmo/time_series/filters.pyx

```diff
@@ -38,25 +38,16 @@ import platform
 
 cimport numpy as np
 import numpy as np
-from libc.math cimport exp, fabs
+from libc.math cimport exp, fabs, isnan
 cimport cython
-
-
-DTYPE_float = np.float
-
-ctypedef np.float_t DTYPE_f
-ctypedef np.double_t DTYPE_d
-ctypedef np.int_t DTYPE_i
-
-cdef extern from "numpy/npy_math.h":
-    bint npy_isnan(double x)
+from cython cimport floating
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-def exp_filter(np.ndarray[DTYPE_d, ndim=1] in_data,
-               np.ndarray[DTYPE_d, ndim=1] in_jd,
+def exp_filter(floating [:] in_data,
```

Comment:
for the exponential filter we actually have another package (pyswi) which is not yet public... we should make pyswi public (currently on gitlab.geo.tuwien.ac.at) and then remove it from pytesmo. @sebhahn and @Adeaem were working on this

Reply:
I only updated the existing code here to not depend on the deprecated numpy API anymore. I would suggest to postpone the removal to another PR, so I can merge this here in soon.

Reply:
sure, we should only remove it when pyswi is public